### ISSUE TITLE
Location searches are underspecified in ODATA

### ISSUE THREAD

**Issue by aaghevli** [opensupporter/osdi-docs#322]
While the ODATA standard has [some thoughts](http://www.odata.org/blog/geospatial-data-support-in-odata/) about how geospacial queries could be done, the spec is currently not complete.

Therefore, in the ResistanceCalendar API, we implemented our own specification which filters by location using coordinates (longitude, latitude), city, or postal code and distance in meters. If coordinates are specified, they are used as they are the preferred method of location based searching.

The following examples may be leveraged in order to standardize location based searches for events. 

http://resistance-calendar.herokuapp.com/v1/events?distance_coords=[-98.435508,29.516496]&distance_max=10000
http://resistance-calendar.herokuapp.com/v1/events?distance_postal_code=94110&distance_max=10000
http://resistance-calendar.herokuapp.com/v1/events?distance_city=Albuquerque&distance_max=10000

**Comment by joshco** [comment]
Thanks for posting this, it's really great to get concrete feedback from implementers on these issues.  This is what we need to define usable solutions.  In it's current state, OSDI has a very meager distance function as part of the filtering specification.

See the "near" function here:
> https://github.com/opensupporter/osdi-docs#filtering-collections

That is something I proposed long ago, but ultimately didn't end up with a customer need to implement it.   I'd be interested, and I think other members would be interested as well, in hearing you talk about your experiences, requirements, and implementation thoughts, and what your recommendations would be.  Perhaps you could drive a proposal based on your work.

Could you contact the chair, @sonyabea to request agenda time on one of our upcoming tech calls?.

**Comment by aaghevli** [comment]
Thanks for the response! 

Oh wow. Not sure how I missed that. I think maybe breaking out the near into nearZipcode, nearLatLong, nearCity may be useful to help server side typing, but that looks good.

I spoke to @sonyabea to share my experiences and they were captured in the meeting notes "OpenSupporter: 11-14-2018" if you care to look, and I am always available for a discussion. 

Thanks again!

**Comment by joshco** [comment]
Process wise, you are correct.  Not only is the discussion in the minutes, but I was actually present at the same meeting.  Not sure how I missed that.

Re the technical aspects, while the OData filter language is very flexible, it's non-trivial to implement a full parser, or one that can handle multiple conditions an functions.

> eg: gender eq ‘Female’ and near(‘10011’, ‘5 miles’)

I looked back at my own code and found that when I needed a quick solution, I used a dedicated query param similar to yours, but for any type.  Maybe we should break location functions outside of the filter string and give them their own params.

As far as server typing, the drawback is that on the client side, where there might be an input field that accepts any type of location, zip, coords, address, the client would have to parse the type, and then stick it in the right query param according to type, as opposed to just passing a string it got along to the server

When I've done this kind of thing with ruby geocoder talking to dstk, google, bing or others, the location API can take any type of location information.  Is your backend different?  What's generating the need to figure out the type?
